src/lightning/fabric/fabric.py

```diff
@@ -678,8 +683,17 @@ def save(self, path: Union[str, Path], state: Dict[str, Union[nn.Module, Optimiz
             path: A path to where the file(s) should be saved
             state: A dictionary with contents to be saved. If the dict contains modules or optimizers, their
                 state-dict will be retrieved and converted automatically.
+            filter: An optional filter that returns a boolean indicating whether the given parameter should be saved
+                (``True``) or filtered out (``False``). It can be a callable that will be applied to each key in
+                ``state`` or a dictionary of the same format as ``state`` mapping each keys to a callable.
```

Comment:
It shouldn't be the keys in the "state". It should be the keys in the model state dict.